### PR TITLE
Bump github.com/emicklei/go-restful/v3 to v3.10.2

### PR DESCRIPTION
**Pull request by lucacome**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Bumps github.com/emicklei/go-restful/v3 from `v3.9.0` to `v3.10.2`

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by apelisse** [comment]
/assign @Jefftree 
Do you want to look at that Jeff?

**Comment by Jefftree** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/115067/pull-kubernetes-verify/1614455634982342656

This test concerns me. Did go restful get updated to append a "/" at the end of every path?

```
-    "/version": {
+    "/version/": {
       "get": {
         "consumes": [
           "application/json"
diff -Naupr -I 'Auto generated by' /home/prow/go/src/k8s.io/kubernetes/api/openapi-spec/v3/.well-known__openid-configuration_openapi.json /home/prow/go/src/k8s.io/kubernetes/_tmp/openapi-spec/v3/.well-known__openid-configuration_openapi.json
--- /home/prow/go/src/k8s.io/kubernetes/api/openapi-spec/v3/.well-known__openid-configuration_openapi.json	2023-01-15 03:43:08.736418744 +0000
+++ /home/prow/go/src/k8s.io/kubernetes/_tmp/openapi-spec/v3/.well-known__openid-configuration_openapi.json	2023-01-15 02:57:49.660892894 +0000
@@ -15,7 +15,7 @@
   },
   "openapi": "3.0.0",
   "paths": {
-    "/.well-known/openid-configuration": {
+    "/.well-known/openid-configuration/": {
       "get": {
         "description": "get service account issuer OpenID configuration, also known as the 'OIDC discovery doc'",
         "operationId": "getServiceAccountIssuerOpenIDConfiguration",
diff -Naupr -I 'Auto generated by' /home/prow/go/src/k8s.io/kubernetes/api/openapi-spec/v3/api__v1_openapi.json /home/prow/go/src/k8s.io/kubernetes/_tmp/openapi-spec/v3/api__v1_openapi.json
--- /home/prow/go/src/k8s.io/kubernetes/api/openapi-spec/v3/api__v1_openapi.json	2023-01-15 03:43:08.856427877 +0000
+++ /home/prow/go/src/k8s.io/kubernetes/_tmp/openapi-spec/v3/api__v1_openapi.json	2023-01-15 02:57:49.664892894 +0000
@@ -9197,7 +9197,7 @@
   },
   "openapi": "3.0.0",
   "paths": {
-    "/api/v1": {
+    "/api/v1/": {
       "get": {
         "description": "get available resources",
         "operationId": "getCoreV1APIResources",
```

**Comment by liggitt** [comment]
I agree we don't really want the observable external change... do we need to adjust usage to preserve existing behavior and openapi?

**Comment by liggitt** [comment]
I think this does belong to api-machinery... it's related to server and endpoint setup

**Comment by lucacome** [comment]
Looks like there's an issue about this https://github.com/emicklei/go-restful/issues/519

File: pkg/kubelet/cri/streaming/server.go [hunk 1]
```diff
@@ -366,9 +366,11 @@ type criAdapter struct {
...
 }
 
-var _ remotecommandserver.Executor = &criAdapter{}
-var _ remotecommandserver.Attacher = &criAdapter{}
-var _ portforward.PortForwarder = &criAdapter{}
+var (
+	_ remotecommandserver.Executor = &criAdapter{}
+	_ remotecommandserver.Attacher = &criAdapter{}
+	_ portforward.PortForwarder    = &criAdapter{}
+)
```
Comment (liggitt) [inline pkg/kubelet/cri/streaming/server.go:373]: this doesn't seem related, suggest dropping it from this PR

File: staging/src/k8s.io/apiserver/pkg/endpoints/discovery/group.go [hunk 2]
```diff
@@ -52,7 +53,7 @@ func NewAPIGroupHandler(serializer runtime.NegotiatedSerializer, group metav1.AP
 func (s *APIGroupHandler) WebService() *restful.WebService {
 	mediaTypes, _ := negotiation.MediaTypesForSerializer(s.serializer)
 	ws := new(restful.WebService)
-	ws.Path(APIGroupPrefix + "/" + s.group.Name)
+	ws.Path(path.Join(APIGroupPrefix, s.group.Name))
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/endpoints/discovery/group.go:56]: is this needed? what was this doing differently previously?

**Review by liggitt — COMMENTED** [review]
this still looks like it's producing diffs in generated openapi paths (pull-kubernetes-verify is failing with diffs in the openapi produced by the server)

**Comment by liggitt** [comment]
/close

in favor of https://github.com/kubernetes/kubernetes/pull/119865 which is rebased
